test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -86,6 +121,97 @@ TEST_F(XdsOverrideHostTest, NoConfigReportsError) {
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
+  ASSERT_NE(picker, nullptr);
+  std::map<UniqueTypeName, std::string> call_attributes{
+      {XdsHostOverrideTypeName(), "no such host"}};
+  std::unordered_set<std::string> picks;
+  for (size_t i = 0; i < kAddresses.size(); ++i) {
+    auto pick = ExpectPickComplete(picker.get(), call_attributes);
+    ASSERT_TRUE(pick.has_value());
+    picks.emplace(*pick);
+  }
+  ASSERT_THAT(picks, UnorderedElementsAreArray(kAddresses));
+}
+
+TEST_F(XdsOverrideHostTest, SubchannelsComeAndGo) {
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
+  ASSERT_NE(picker, nullptr);
+  // Make sure child policy works
+  EXPECT_NE(ExpectPickComplete(picker.get()), ExpectPickComplete(picker.get()));
+  // Check that the host is overridden
+  std::map<UniqueTypeName, std::string> call_attributes{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  // Some other address is gone
+  EXPECT_EQ(
+      ApplyUpdate(BuildUpdate({"ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442"},
+                              MakeXdsOverrideHostConfig("round_robin")),
+                  policy_.get()),
+      absl::OkStatus());
+  ExpectState(GRPC_CHANNEL_READY);
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  // "Our" address is gone
+  EXPECT_EQ(
+      ApplyUpdate(BuildUpdate({"ipv4:127.0.0.1:441", "ipv4:127.0.0.1:443"},
+                              MakeXdsOverrideHostConfig("round_robin")),
+                  policy_.get()),
+      absl::OkStatus());
+  ExpectState(GRPC_CHANNEL_READY);
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ExpectQueueEmpty();
+  ASSERT_NE(picker, nullptr);
+  std::array<absl::optional<std::string>, 2> picks = {
+      ExpectPickComplete(picker.get(), call_attributes),
+      ExpectPickComplete(picker.get(), call_attributes)};
+  EXPECT_THAT(picks, UnorderedElementsAreArray(
+                         {"ipv4:127.0.0.1:441", "ipv4:127.0.0.1:443"}));
+  // And now it is back
+  EXPECT_EQ(
+      ApplyUpdate(BuildUpdate({"ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"},
+                              MakeXdsOverrideHostConfig("round_robin")),
+                  policy_.get()),
+      absl::OkStatus());
+  ExpectState(GRPC_CHANNEL_READY);
```

Comment:
Same as above: Please use `WaitForRoundRobinListChange()` to wait for the policy to switch to the new address list.